pkg/cloud/services/iamauth/reconcile.go

```diff
@@ -21,23 +21,17 @@ import (
 	"fmt"
 
 	"github.com/aws/aws-sdk-go/aws"
-	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/aws/aws-sdk-go/service/iam"
 	"github.com/pkg/errors"
 	"k8s.io/klog/v2"
 
 	ekscontrolplanev1 "sigs.k8s.io/cluster-api-provider-aws/v2/controlplane/eks/api/v1beta2"
-	iamv1 "sigs.k8s.io/cluster-api-provider-aws/v2/iam/api/v1beta1"
 )
 
 // ReconcileIAMAuthenticator is used to create the aws-iam-authenticator in a cluster.
-func (s *Service) ReconcileIAMAuthenticator(ctx context.Context) error {
+func (s *Service) ReconcileIAMAuthenticator(ctx context.Context, nodeRoles []string) error {
```

Comment:
```suggestion
func (s *Service) ReconcileIAMAuthenticator(ctx context.Context, nodeRoles map[string]struct{}) error {
```

Reply:
Thanks for the suggestion. I made the changes